azure-client-runtime/src/main/java/com/microsoft/azure/v2/AzureProxy.java

```diff
@@ -0,0 +1,112 @@
+package com.microsoft.azure.v2;
+
+import com.microsoft.rest.RestException;
+import com.microsoft.rest.protocol.SerializerAdapter;
+import com.microsoft.rest.v2.RestProxy;
+import com.microsoft.rest.v2.SwaggerMethodParser;
+import com.microsoft.rest.v2.http.HttpClient;
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Observable;
+import rx.Single;
+import rx.functions.Func1;
+
+import java.io.IOException;
+
+public class AzureProxy {
+    /**
+     * Create a proxy implementation of the provided Swagger interface.
+     * @param swaggerInterface The Swagger interface to provide a proxy implementation for.
+     * @param httpClient The internal HTTP client that will be used to make REST calls.
+     * @param serializer The serializer that will be used to convert POJOs to and from request and
+     *                   response bodies.
+     * @param <A> The type of the Swagger interface.
+     * @return A proxy implementation of the provided Swagger interface.
+     */
+    @SuppressWarnings("unchecked")
+    public static <A> A create(Class<A> swaggerInterface, final HttpClient httpClient, SerializerAdapter<?> serializer) {
+        return RestProxy.create(swaggerInterface, httpClient, serializer, new RestProxy.ResponseHandler() {
+            @Override
+            public Object handleSyncResponse(HttpResponse response, SwaggerMethodParser methodParser, SerializerAdapter<?> serializer) throws IOException, RestException {
+                while (response.statusCode() == 202) {
+                    final String location = response.headerValue("Location");
+                    final HttpRequest pollRequest = new HttpRequest(methodParser.fullyQualifiedMethodName(), "GET", location);
+                    response = httpClient.sendRequest(pollRequest);
```

Comment:
Wait for the number of seconds denoted by the Retry-After response header.
https://github.com/Azure/autorest-clientruntime-for-java/blob/1abab34b13adf2d7dba7e5ecc73b376e6f7a113e/azure-client-runtime/src/main/java/com/microsoft/azure/PollingState.java#L426

Probably a lot of the little details can be recreated from this class, or this class will prove to still be useful after OkHttp/Retrofit has been factored out of it.

Reply:
Added basic support for Retry-After. I think it'll be smart to go through PollingState and to create unit tests for each of the scenarios that it accounts for. Then I can easily go through AzureProxy and make sure all of the cases are covered.